src/main/java/de/rub/nds/anvilcore/junit/extension/AnvilTestWatcher.java

```diff
@@ -382,4 +384,61 @@ private String retrieveThrowableReason(Throwable thrown) {
             return stringWriter.toString();
         }
     }
+
+    /**
+     * Logs a summary of all test runs and in case of failure, the details of the failed test cases.
+     *
+     * @param context AnvilContext containing all test runs and their results
+     */
+    public void logTestPlanExecutionSummary(AnvilContext context) {
+        StringBuilder logMessage = new StringBuilder("\nTest plan execution summary:");
+        context.getResultsTestRuns()
+                .forEach(
+                        (testRunResult, testRuns) -> {
+                            logMessage.append(
+                                    String.format(
+                                            "\n\t%d test runs %s", testRuns.size(), testRunResult));
+
+                            logMessage.append(
+                                    buildTestCaseFailureDetailsSummary(testRunResult, testRuns));
+                        });
+
+        LOGGER.info(logMessage.toString());
```

Comment:
I am unsure about the best way to implement this sorting feature.

A readable implementation from a developer's point of view would be a simple compare for the ScorePercentage of the TestResult:

```java
context.getResultsTestRuns().entrySet().stream()
                .sorted(
                        Map.Entry.comparingByKey(
                                Comparator.comparing(TestResult::getScorePercentage).reversed()))
                ...
```

But this approach has one major drawback: there are multiple TestResults with ScorePercentage == 0, including FULLY_FAILED, so that the order of the summaries of test executions with different test runs (and so, different test results) will differ:

```java
public enum TestResult {
    NOT_SPECIFIED(1, 0),
    STRICTLY_SUCCEEDED(1 << 1, 100),
    CONCEPTUALLY_SUCCEEDED(1 << 2, 80),
    PARTIALLY_FAILED(1 << 3, 20),
    FULLY_FAILED(1 << 4, 0),
    DISABLED(1 << 5, 0),
    TEST_SUITE_ERROR(1 << 6, 0);
```

However, the alternative, which treats the individual test results with ScorePercentage == 0 separately, would increase the complexity of the compare operation, which is why I am unsere whether this is a good way to go.

Reply:
Why not just use `getValue` for sorting? The order should be fine except for NOT_SPECIFIED, which we have to fix anyway

Reply:
I assumed that the values are merely a loose coupling to the name of the TestResult and that their order is arbitrary, i.e. not fixed. I have made this assumption because there is no explicit reference to a fixed order.

If someone changes these values later as part of a refactoring and also changes the order, this would have consequences for the sorting of the test execution summary that would hardly be noticed during the refactoring.

Reply:
These bit shifts are more or less deprecated code. Maybe we add a custom comparator to the enum which defines the order?

Reply:
I implemented a custom comparator that compares based on the TestResult value in commit [ef31849](https://github.com/tls-attacker/Anvil-Core/pull/31/commits/ef3184914ada1b0b98a9ac860a9b2a34961e311e)